deps/gen_constants.f90

```diff
@@ -54,6 +54,7 @@ program gen_constants
   call output("MPI_ANY_TAG     ", MPI_ANY_TAG)
   call output("MPI_TAG_UB      ", MPI_TAG_UB)
   call output("MPI_UNDEFINED   ", MPI_UNDEFINED)
+  call output("MPI_IN_PLACE    ", MPI_IN_PLACE)
```

Comment:
(See comment below.) This should probably output `loc(MPI_IN_PLACE)` instead. `loc` is a Fortran extension that is supported by most compilers -- I'd guess that we can rely on it. We also need to output it as pointer-sized value, not as integer, so calling `output` might not work. (This is equivalent to using the `&` address-of operator in C.)